bookdown/07-special-cluster.Rmd

```diff
@@ -0,0 +1,178 @@
+In hierarchical clustering, the goal is to build a hierarchy of nested clusters by either splitting large clusters into smaller ones or merging smaller clusters into bigger ones.
+The final result is a tree or dendrogram which can change if a new data point is added.
+For consistency with the rest of the ecosystem, `r mlr_pkg("mlr3cluster")` offers `predict` method for hierarchical clusterers but it simply assigns all points to a specified number of clusters by cutting the resulting tree at a corresponding level.
+Moreover, some learners estimate the probability of each observation belonging to a given cluster.
+`predict_types` field gives a list of prediction types for each learner.
+
+After training, the `model` field stores a learner's model that looks different for each learner depending on the underlying library.
+`predict` returns a `PredictionClust` object that gives a simplified view of the learned model.
+If the data given to the `predict` method is the same as the one on which the learner was trained, `predict` simply returns cluster assignments for the "training" observations.
+On the other hand, if the test set contains new data, `predict` will estimate cluster assignments for that data set.
+Some learners do not support estimating cluster partitions on new data and will instead return assignments for training data and print a warning message.
+
+In the following example, a `r ref("mlr_learners_clust.kmeans", text = "$k$-means learner")` is applied on the `r ref("mlr_tasks_usarrests", text = "US arrest data set")`.
+The class labels are predicted and the contribution of the task features to assignment of the respective class are visualized.
+```{r 07-special-cluster-002, message=FALSE, warning=FALSE}
+library("mlr3")
+library("mlr3cluster")
+library("mlr3viz")
+set.seed(1L)
+
+# create an example task
+task = tsk("usarrests")
+print(task)
+autoplot(task)
+
+# create a k-means learner
+learner = lrn("clust.kmeans")
+
+# assigning each observation to one of the two clusters (default in clust.kmeans)
+learner$train(task)
+learner$model
+
+# make "predictions" for the same data
+prediction = learner$predict(task)
+autoplot(prediction, task)
+```
+
+
+### Measures
+The difference between supervised and unsupervised learning is that there is no ground truth data in unsupervised learning.
+In a supervised setting, such as classification, we would need to compare our predictions to true labels.
+Since clustering is an example of unsupervised learning, there are no true labels to which we can compare.
+However, we can still measure the quality of cluster assignments by quantifying how closely objects within the same cluster are related (cluster cohesion) as well as how distinct different clusters are from each other (cluster separation).
+
+To assess the quality of clustering, there are a few built-in evaluation metrics available.
+One of them is within sum of squares which quantifies cluster cohesion.
```

Comment:
Sum of squares of what? Distances?

Reply:
I added more info on the measure. Thank you!